### PR TITLE
#17612 Fixing performance problems

### PR DESCRIPTION
**Pull request by freddyucv**
Fixing: https://github.com/dotCMS/core/issues/17612#issuecomment-570007342

The error was here:

https://github.com/dotCMS/core/pull/17759/files#diff-addcedfda0cc830f3c7ced2a64173641R117

when the Optional was empty then the get method throw an Exception, then the default host was not catching.

Really we have two methods to get the current Host in different API:

- https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java#L64
- https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPIImpl.java#L364

so the change was remoce the method from HTMLPageAssetRenderedAPIImpl

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java [hunk 1]
```diff
@@ -52,49 +57,97 @@ public Host getHost(HttpServletRequest request)  {
 
     @CloseDBIfOpened
     public Host getCurrentHost(final HttpServletRequest request)
-            throws DotDataException, DotSecurityException, PortalException, SystemException {
+            throws DotDataException, DotSecurityException {
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java]: this exception really are not thrown

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,719 @@
...
+     * Should: host_id parameter be take over any other source
+     */
+    @Test
+    public void useHostIDRequestAsCurrentHost() throws DotDataException, DotSecurityException {
+        final Host host = new SiteDataGen().nextPersisted();
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        final HttpSession session = mock(HttpSession.class);
+
+        setAllParametersAndAttributes(request, session);
+        when(request.getParameter("host_id")).thenReturn(host.getIdentifier());
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [The String literal "host_id" appears 15 times in this file; the first occurrence is on line 46](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,719 @@
...
+
+        final User user = createBackendUser();
+
+        final HostWebAPIImpl hostWebAPI = new HostWebAPIImpl();
+
+        try {
+            hostWebAPI.getCurrentHost(request, user);
+            throw new RuntimeException("should throw a DotSecurityException");
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [The String literal "should throw a DotSecurityException" appears 5 times in this file; the first occurrence is on line 105](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,719 @@
...
+        when(session.getAttribute(WebKeys.CURRENT_HOST)).thenReturn(null);
+        when(session.getAttribute(WebKeys.CMS_SELECTED_HOST_ID)).thenReturn(host.getIdentifier());
+
+        final User user = createBackendUser();
+
+        final HostWebAPIImpl hostWebAPI = new HostWebAPIImpl();
+
+        try {
+            hostWebAPI.getCurrentHost(request, user);
+            throw new RuntimeException("should throw a DotSecurityException");
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java [hunk 5]
```diff
@@ -52,49 +57,97 @@ public Host getHost(HttpServletRequest request)  {
...
+
+        if (session == null){
+            return Optional.empty();
+        }
+
+        Host host = null;
+        if (session.getAttribute(WebKeys.CURRENT_HOST) != null) {
+            final Host hostFromSession = (Host) session.getAttribute(WebKeys.CURRENT_HOST);
+            final Object hostId = session.getAttribute(WebKeys.CMS_SELECTED_HOST_ID);
+            host = hostFromSession.getIdentifier().equals(hostId) ? hostFromSession : null;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java]: Codacy Issue found: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java [hunk 6]
```diff
@@ -52,49 +57,97 @@ public Host getHost(HttpServletRequest request)  {
...
 
+        if (session != null && user.isBackendUser()) {
             session.setAttribute(WebKeys.CURRENT_HOST, host);
         }
+    }
 
-        return host;
+    private void checkHostPermission(User user, boolean respectAnonPerms, Host host) throws DotDataException, DotSecurityException {
+        if(!APILocator.getPermissionAPI().doesUserHavePermission(host, PermissionAPI.PERMISSION_READ, user, respectAnonPerms)){
+            String u = (user != null) ? user.getUserId() : null;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java]: Codacy Issue found: [Local variable 'u' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,719 @@
...
+        final HttpSession session = mock(HttpSession.class);
+        when(request.getSession(false)).thenReturn(session);
+        when(request.getServerName()).thenReturn("not_exists_host");
+
+        final User user = createBackendUser();
+        final HostWebAPIImpl hostWebAPI = new HostWebAPIImpl();
+
+        try {
+            hostWebAPI.getCurrentHost(request, user);
+        }catch(DotSecurityException e) {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid empty catch blocks](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPI.java [hunk 8]
```diff
@@ -28,7 +31,28 @@
...
+	 *
+	 * @param request current {@link HttpServletRequest}
+	 * @param user User current login user
+	 * @return
+	 * @throws DotDataException
+	 * @throws DotSecurityException
+	 * @throws PortalException
+	 * @throws SystemException
+	 */
+	public Host getCurrentHost(final HttpServletRequest request, final User user) throws DotDataException, DotSecurityException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPI.java]: Codacy Issue found: [Unnecessary modifier 'public' on method 'getCurrentHost': the method is declared in an interface type](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java [hunk 9]
```diff
@@ -52,49 +57,97 @@ public Host getHost(HttpServletRequest request)  {
...
+
+        return Optional.ofNullable(host);
+    }
+
+    private Optional<Host> getCurrentHostFromRequest(final HttpServletRequest request, final User user, boolean respectAnonPerms)
+            throws DotDataException, DotSecurityException {
+
+	    final String hostId = request.getParameter("host_id");
+
+	    if ((hostId != null && user.isBackendUser())) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java [hunk 10]
```diff
@@ -7,11 +7,15 @@
 import com.dotcms.repackage.javax.portlet.ActionRequest;
 import com.dotcms.repackage.javax.portlet.RenderRequest;
 import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
 import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.PermissionLevel;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.business.PermissionLevel'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java [hunk 11]
```diff
@@ -52,49 +57,97 @@ public Host getHost(HttpServletRequest request)  {
...
         request.setAttribute(WebKeys.CURRENT_HOST, host);
-        if (session != null && mode.isAdmin) {
 
+        if (session != null && user.isBackendUser()) {
             session.setAttribute(WebKeys.CURRENT_HOST, host);
         }
+    }
 
-        return host;
+    private void checkHostPermission(User user, boolean respectAnonPerms, Host host) throws DotDataException, DotSecurityException {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java]: Codacy Issue found: [Parameter 'user' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java [hunk 12]
```diff
@@ -52,49 +57,97 @@ public Host getHost(HttpServletRequest request)  {
...
+            String u = (user != null) ? user.getUserId() : null;
+
+            String message = "User " + u + " does not have permission to host:" + host.getHostname();
+            Logger.error(HostAPIImpl.class, message);
+            throw new DotSecurityException(message);
+        }
     }
 
-	@Override
+    private Optional<Host> getCurrentHostFromSession(final HttpServletRequest request, User user, boolean respectAnonPerms)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java]: Codacy Issue found: [Parameter 'user' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java [hunk 13]
```diff
@@ -0,0 +1,719 @@
...
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Permission;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import static junit.framework.TestCase.assertEquals;
+import static org.mockito.Mockito.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.mockito.Mockito'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java]: Codacy Issue found: [Avoid variables with short names like u](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java [hunk 14]
```diff
@@ -0,0 +1,719 @@
...
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+public class HostWebAPIImplIntegrationTest extends IntegrationTestBase {
+
+    @BeforeClass
+    public static void prepare () throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java [hunk 15]
```diff
@@ -52,49 +57,97 @@ public Host getHost(HttpServletRequest request)  {
...
+
+        if (host == null && session.getAttribute(WebKeys.CMS_SELECTED_HOST_ID) != null) {
+            final String hostId = (String) session.getAttribute(WebKeys.CMS_SELECTED_HOST_ID);
+            host = find(hostId, user, respectAnonPerms);
+        }
+
+        return Optional.ofNullable(host);
+    }
+
+    private Optional<Host> getCurrentHostFromRequest(final HttpServletRequest request, final User user, boolean respectAnonPerms)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java]: Codacy Issue found: [Parameter 'respectAnonPerms' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java [hunk 16]
```diff
@@ -52,49 +57,97 @@ public Host getHost(HttpServletRequest request)  {
...
             session.setAttribute(WebKeys.CURRENT_HOST, host);
         }
+    }
 
-        return host;
+    private void checkHostPermission(User user, boolean respectAnonPerms, Host host) throws DotDataException, DotSecurityException {
+        if(!APILocator.getPermissionAPI().doesUserHavePermission(host, PermissionAPI.PERMISSION_READ, user, respectAnonPerms)){
+            String u = (user != null) ? user.getUserId() : null;
+
+            String message = "User " + u + " does not have permission to host:" + host.getHostname();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java]: Codacy Issue found: [Local variable 'message' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

**Review by jdotcms — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPI.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPI.java]: Codacy Issue found: [Unnecessary modifier 'public' on method 'getCurrentHost': the method is declared in an interface type](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java [hunk 17]
```diff
@@ -0,0 +1,721 @@
...
+import com.liferay.portal.model.User;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+public class HostWebAPIImplIntegrationTest extends IntegrationTestBase {
+
+    public static final String HOST_ID_PARAMETER_NAME = "host_id";
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [The String literal "host_id" appears 15 times in this file; the first occurrence is on line 29](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java [hunk 18]
```diff
@@ -0,0 +1,721 @@
...
+
+    @BeforeClass
+    public static void prepare () throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java [hunk 19]
```diff
@@ -0,0 +1,721 @@
...
+        setAllParametersAndAttributes(request, session);
+        when(request.getParameter("host_id")).thenReturn(host.getIdentifier());
+
+        final User user = createBackendUser();
+
+        final HostWebAPIImpl hostWebAPI = new HostWebAPIImpl();
+
+        try {
+            hostWebAPI.getCurrentHost(request, user);
+            throw new RuntimeException("should throw a DotSecurityException");
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [The String literal "should throw a DotSecurityException" appears 5 times in this file; the first occurrence is on line 105](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java [hunk 20]
```diff
@@ -0,0 +1,721 @@
...
+        final HttpSession session = mock(HttpSession.class);
+        when(request.getSession(false)).thenReturn(session);
+        when(request.getServerName()).thenReturn("not_exists_host");
+
+        final User user = createBackendUser();
+        final HostWebAPIImpl hostWebAPI = new HostWebAPIImpl();
+
+        try {
+            hostWebAPI.getCurrentHost(request, user);
+        }catch(DotSecurityException e) {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid empty catch blocks](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java [hunk 21]
```diff
@@ -52,49 +54,99 @@ public Host getHost(HttpServletRequest request)  {
...
         request.setAttribute(WebKeys.CURRENT_HOST, host);
-        if (session != null && mode.isAdmin) {
 
+        if (session != null && user.isBackendUser()) {
             session.setAttribute(WebKeys.CURRENT_HOST, host);
         }
+    }
 
-        return host;
+    private void checkHostPermission(final User user, boolean respectAnonPerms, final Host host)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java]: Codacy Issue found: [Parameter 'respectAnonPerms' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java [hunk 22]
```diff
@@ -0,0 +1,721 @@
...
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Permission;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import static junit.framework.TestCase.assertEquals;
+import static org.mockito.Mockito.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.mockito.Mockito'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java [hunk 23]
```diff
@@ -52,49 +54,99 @@ public Host getHost(HttpServletRequest request)  {
...
+
+        if (session == null){
+            return Optional.empty();
+        }
+
+        Host host = null;
+        if (session.getAttribute(WebKeys.CURRENT_HOST) != null) {
+            final Host hostFromSession = (Host) session.getAttribute(WebKeys.CURRENT_HOST);
+            final Object hostId = session.getAttribute(WebKeys.CMS_SELECTED_HOST_ID);
+            host = hostFromSession.getIdentifier().equals(hostId) ? hostFromSession : null;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java]: Codacy Issue found: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java [hunk 24]
```diff
@@ -0,0 +1,721 @@
...
+        when(session.getAttribute(WebKeys.CURRENT_HOST)).thenReturn(null);
+        when(session.getAttribute(WebKeys.CMS_SELECTED_HOST_ID)).thenReturn(host.getIdentifier());
+
+        final User user = createBackendUser();
+
+        final HostWebAPIImpl hostWebAPI = new HostWebAPIImpl();
+
+        try {
+            hostWebAPI.getCurrentHost(request, user);
+            throw new RuntimeException("should throw a DotSecurityException");
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java [hunk 25]
```diff
@@ -0,0 +1,721 @@
...
+     */
+    @Test
+    public void useHostIDRequestAsCurrentHostNotBackendUser() throws DotDataException, DotSecurityException {
+        final Host host = new SiteDataGen().nextPersisted();
+        final Host anotherHost = new SiteDataGen().nextPersisted();
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        final HttpSession session = mock(HttpSession.class);
+
+        setAllParametersAndAttributes(request, session, anotherHost);
+        when(request.getParameter("host_id")).thenReturn(host.getIdentifier());
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Use the constant `HOST_ID_PARAMETER_NAME`

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java [hunk 26]
```diff
@@ -0,0 +1,721 @@
...
+     * Should: throw a {@link DotSecurityException}
+     */
+    @Test
+    public void useHostIdAndTheUserDontHavePermission() throws DotDataException, DotSecurityException {
+        final Host host = new SiteDataGen().nextPersisted();
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        final HttpSession session = mock(HttpSession.class);
+
+        setAllParametersAndAttributes(request, session);
+        when(request.getParameter("host_id")).thenReturn(host.getIdentifier());
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Use the constant `HOST_ID_PARAMETER_NAME`

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java [hunk 27]
```diff
@@ -0,0 +1,721 @@
...
+        when(session.getAttribute(WebKeys.CURRENT_HOST)).thenReturn(null);
+        when(session.getAttribute(WebKeys.CMS_SELECTED_HOST_ID)).thenReturn(host.getIdentifier());
+
+        final User user = createBackendUser();
+
+        final HostWebAPIImpl hostWebAPI = new HostWebAPIImpl();
+
+        try {
+            hostWebAPI.getCurrentHost(request, user);
+            throw new RuntimeException("should throw a DotSecurityException");
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java [hunk 28]
```diff
@@ -0,0 +1,721 @@
...
+        final HttpSession session = mock(HttpSession.class);
+        when(request.getSession(false)).thenReturn(session);
+        when(request.getServerName()).thenReturn("not_exists_host");
+
+        final User user = createBackendUser();
+        final HostWebAPIImpl hostWebAPI = new HostWebAPIImpl();
+
+        try {
+            hostWebAPI.getCurrentHost(request, user);
+        }catch(DotSecurityException e) {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid empty catch blocks](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java [hunk 29]
```diff
@@ -0,0 +1,721 @@
...
+        setAllParametersAndAttributes(request, session);
+        when(request.getParameter(HOST_ID_PARAMETER_NAME)).thenReturn(host.getIdentifier());
+
+        final User user = createBackendUser();
+
+        final HostWebAPIImpl hostWebAPI = new HostWebAPIImpl();
+
+        try {
+            hostWebAPI.getCurrentHost(request, user);
+            throw new RuntimeException("should throw a DotSecurityException");
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [The String literal "should throw a DotSecurityException" appears 5 times in this file; the first occurrence is on line 105](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPI.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPI.java]: Codacy Issue found: [Unnecessary modifier 'public' on method 'getCurrentHost': the method is declared in an interface type](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java (hunk 18 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java (hunk 22 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.mockito.Mockito'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java (hunk 21 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java]: Codacy Issue found: [Parameter 'respectAnonPerms' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java (hunk 23 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java]: Codacy Issue found: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java (hunk 23 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java]: Codacy Issue found: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java (hunk 27 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java (hunk 18 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java (hunk 28 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid empty catch blocks](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java (hunk 21 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java]: Codacy Issue found: [Parameter 'respectAnonPerms' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPI.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPI.java]: Codacy Issue found: [Unnecessary modifier 'public' on method 'getCurrentHost': the method is declared in an interface type](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java (hunk 29 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [The String literal "should throw a DotSecurityException" appears 5 times in this file; the first occurrence is on line 105](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

File: dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java (hunk 22 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/web/HostWebAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.mockito.Mockito'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)

**Comment by freddyucv** [comment]
Note for QA:

- Test that the bug describe here

https://github.com/dotCMS/core/issues/17612

in the "Steps to reproduce the behavior" section is not happening anymore.

And also that when a request is done twice with a front end user, in the second chance it should take the default host directly from the cache.

**Pull request merged**